src/debugger/debug_workflow.ts

```diff
@@ -121,29 +120,38 @@ export class WorkflowContextDebug extends DBOSContextImpl implements WorkflowCon
       },
       this.span
     );
-    let check: RecordedResult<R>;
+    let check: RecordedResult<R> | Error;
 
     const wrappedTransaction = async (client: UserDatabaseClient): Promise<R> => {
       // Original result must exist during replay.
       const tCtxt = new TransactionContextImpl(this.#dbosExec.userDatabase.getName(), client, this, span, this.#dbosExec.logger, funcID, txn.name);
       check = await this.checkExecution<R>(client, funcID);
 
+      if (check instanceof Error) {
+        this.logger.warn(`original transaction ${txn.name} failed with error: ${check.message}`);
+      }
+
       // Execute the user's transaction.
       const result = await txn(tCtxt, ...args);
       return result;
     };
 
     const result = await this.#dbosExec.userDatabase.transaction(wrappedTransaction, txnInfo.config);
```

Comment:
We might need to try and catch the error thrown by the transaction here, and always re-thrown the original error in case the following code in the workflow has a conditional branch that depends on this error.

The reason is that an errored transactions may throw a different error when we replay it, especially if the original error was caused by some non-deterministic DB schedules.